test/integration/kubernetes_deploy_test.rb

```diff
@@ -550,6 +550,32 @@ def test_success_detection_tolerates_out_of_band_deployment_scaling
     assert_equal true, success, "Failed to deploy deployment with dynamic replica count"
   end
 
+  def test_successful_daemon_set_deploy
+    assert deploy_fixtures("hello-cloud", subset: ["daemon_set.yml"])
+
+    assert_logs_match_all([
+      'Successfully deployed 1 resource',
+      '1 currentNumberSchedule, 1 desiredNumberSchedule, 1 numberRead'
+    ])
+  end
+
+  def test_timed_out_daemon_set_deploy
```

Comment:
Since the timeout has not special logic for this resource type, I'd rather swap this out for a test of a failure scenario. There are a few deployment tests you can use as a starting point (no need to duplicate all of them, since they're ultimately using the same pod logic). One thing it is important to show is that we're successfully fetching events from both the DS and a pod, as well as logs.